Content.Shared/RCD/Components/RCDComponent.cs

```diff
@@ -25,27 +15,35 @@ public enum RcdMode : byte
 public sealed partial class RCDComponent : Component
 {
     /// <summary>
-    /// Time taken to do an action like placing a wall
+    /// List of RCD prototypes that the device comes loaded with
     /// </summary>
-    [DataField("delay"), ViewVariables(VVAccess.ReadWrite), AutoNetworkedField]
-    public float Delay = 2f;
+    [DataField, AutoNetworkedField, ViewVariables(VVAccess.ReadWrite)]
+    public HashSet<ProtoId<RCDPrototype>> AvailablePrototypes = new();
 
-    [DataField("swapModeSound")]
-    public SoundSpecifier SwapModeSound = new SoundPathSpecifier("/Audio/Items/genhit.ogg");
-
-    [DataField("successSound")]
+    /// <summary>
+    /// Sound that plays when a RCD operation successfully completes
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
     public SoundSpecifier SuccessSound = new SoundPathSpecifier("/Audio/Items/deconstruct.ogg");
 
     /// <summary>
-    /// What mode are we on? Can be floors, walls, airlock, deconstruct.
+    /// The ProtoId of the currently selected RCD prototype
+    /// </summary>
+    [AutoNetworkedField, ViewVariables(VVAccess.ReadOnly)]
+    public ProtoId<RCDPrototype> ProtoId = default!;
+
```

Comment:
Datafield?

Reply:
I don't want it to be set directly because I want to make sure that the cached prototype gets updated